src/Trakx.CoinGecko.ApiClient/Clients/CoinGeckoClient_Internal.cs

```diff
@@ -218,6 +238,8 @@ private static (string BaseIds, string QuoteIds) GetIdsForPriceQuery(
 
         vsCurrencies ??= [];
 
+        var supportedQuoteCurrencies = await GetSupportedQuoteCurrencies(cancellationToken);
```

Comment:
This was being done by each caller, might as well move inside the common method.